contracts/Builders/SimpleBuilder.sol

```diff
@@ -65,15 +46,25 @@ contract SimpleBuilder is ERC721Holder {
         assert(totalAmount == 0);
     }
 
+    function _createFirstNFT(
+        ISimpleProvider provider,
+        address token,
+        uint256 totalAmount,
+        UserPool calldata userData,
+        uint256[] calldata params
+    ) internal validUserData(userData) returns (uint256 poolId, uint256 amount) {
+        poolId = lockDealNFT.mintAndTransfer(userData.user, token, msg.sender, totalAmount, provider);
+        provider.registerPool(poolId, _concatParams(userData.amount, params));
```

Comment:
lets use `_concatParams` only 1 time, and just swap params[0] for userData.amount